testing/endtoend/components/validator.go

```diff
@@ -372,3 +383,35 @@ func FeeRecipientFromPubkey(key string) string {
 	// pubkey[:(2+fieldparams.FeeRecipientLength*2)] slicing 2 (for the 0x preamble) + 2 hex chars for each byte
 	return common.HexToAddress(key[:(2 + fieldparams.FeeRecipientLength*2)]).Hex()
 }
+
+// writeLinesToFile writes a slice of strings to a file, each string on a new line.
+func writeLinesToFile(lines []string, filename string) error {
+	// Open the file for writing. If the file does not exist, create it, or truncate it if it does.
+	f, err := os.Create(filepath.Clean(filename))
+	if err != nil {
+		return fmt.Errorf("error creating file: %w", err)
+	}
+	defer func(file *os.File) {
+		err := file.Close()
+		if err != nil {
+			log.Error(err.Error())
+		}
+	}(f)
+
+	// Create a buffered writer from the file
+	writer := bufio.NewWriter(f)
```

Comment:
Why are you creating a buffered writer? Seems like you could just open the file, call WriterString for each line, then close it - no extra buffer, no flushing.